File: 1846-maximum-element-after-decreasing-and-rearranging/1846-maximum-element-after-decreasing-and-rearranging.py
class Solution(object):
    def maximumElementAfterDecrementingAndRearranging(self, arr):
        arr.sort()
        arr[0]=1
        max_val=1
        for i in range(1,len(arr)):
           if arr[i]-arr[i-1]>1:
            arr[i]=arr[i-1]+1
           max_val=max(max_val,arr[i])
        return max_val
        

File: 1846-maximum-element-after-decreasing-and-rearranging/1846-maximum-element-after-decreasing-and-rearranging.java
class Solution {
    public int maximumElementAfterDecrementingAndRearranging(int[] arr) {
        Arrays.sort(arr);
        arr[0]=1;
        int max=1;
        int n=arr.length;
        for(int i=1;i<arr.length;i++){
                if((arr[i]-arr[i-1])>1){
                    arr[i]=arr[i-1]+1;
                }
               max= Math.max(arr[i],max);
        }
        
return max;
    }
}